### PR TITLE
[test] Improve question list validation

### PR DESCRIPTION
**Pull request by Drarig29**
# What this PR is about?

This PR improves the validation that was only done on the `img_path` property. I've started to see PRs with wrong slugs, that's why I had the idea to validate them: they must be lowercase, with only alphanumeric characters and em dashes (no underscore or uppercase letter).

I also added validation for the left/right choice's titles, so that the text is always perfectly centered.

It also fixes some typos and inconsistencies.

### Tests

- [x] The `isSlugValid()` function can self-test itself against known invalid and valid examples. The test is run when the function is exported.

### Clean Code

- [x] I made sure the code is type safe (no any)

### CODE REVIEW

File: scripts/preview-generator.ts [hunk 1]
```diff
@@ -38,31 +40,45 @@ const renderPreviewBySlug = async (slug: string, override = false) => {
     return
   }
 
-  if (question.choiceLeft.img_path.indexOf('webp') > -1 || question.choiceRight.img_path.indexOf('webp') > -1) {
+  if (question.choiceLeft.img_path.endsWith('.webp') || question.choiceRight.img_path.endsWith('.webp')) {
```
Comment (bdebon) [inline scripts/preview-generator.ts:43]: I did not know the `endsWith` method ! Nice !!

**Review by bdebon — APPROVED** [review]
Amazing contribution! Thank you a lot for cleaning and making all of this more robust. I appreciate a lot!

File: scripts/preview-generator.ts (hunk 1 above)
Comment (Drarig29) [reply scripts/preview-generator.ts:43]: Happy to teach you something! 😁

**Pull request merged**